context/app/static/js/components/entity-search/sidebar/DatasetConfigureSearchTable/hooks.js

```diff
@@ -0,0 +1,96 @@
+import produce from 'immer';
+
+import { useSelectedItems } from 'js/components/entity-search/sidebar/ConfigureSearch/hooks';
+import { createDatasetFacet } from 'js/components/entity-search/SearchWrapper/utils';
+import fieldsToAssayMap from 'metadata-field-assays';
+import { useStore } from 'js/components/entity-search/SearchWrapper/store';
+import { invertKeyToArrayMap as createDataTypesToFieldsMap } from './utils';
+import { useMetadataFieldConfigs } from '../ConfigureSearchTable/hooks';
+
+const dataTypesToFieldsMap = createDataTypesToFieldsMap(fieldsToAssayMap);
+
+function selectedItemsReducer(state, { type, payload }) {
+  const tempState = state;
+  switch (type) {
+    case 'selectItem':
+      return { ...state, [payload]: undefined }; // using an object as a set
```

Comment:
Not clear about this... but I'm not clear on the bigger picture either.

Reply:
A set would make sense here, but updating a set doesn't trigger react updates. There are ways around it by redefining a new set each time, but it adds more code so using an object  here where the values don't matter made sense.